packages/plugins/i18n/server/src/services/ai-localizations.ts

```diff
@@ -138,6 +135,31 @@ const createAILocalizationsService = ({ strapi }: { strapi: Core.Strapi }) => {
         });
       }
 
+      /**
+       * Provide a schema to the LLM so that we can give it instructions about how to handle each
+       * type of attribute. Only keep essential schema data to avoid cluttering the context.
+       * Ignore fields that don't need to be localized.
+       * TODO: also provide a schema of all the referenced components
+       */
+      const minimalContentTypeSchema = Object.fromEntries(
+        Object.entries(schema.attributes)
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          .filter(([_, attr]) => {
+            const isLocalized = isLocalizedAttribute(attr);
+            const isSupportedType = !UNSUPPORTED_ATTRIBUTE_TYPES.includes(attr.type);
+            return isLocalized && isSupportedType;
+          })
+          .map(([key, attr]) => {
+            const minimalAttribute = { type: attr.type };
+            if (attr.type === 'component') {
```

Comment:
Not sure if it's useful but the utils package exports helpers for components and dynamic zones

https://github.com/strapi/strapi/blob/662e18309746550b356023ea04a5fb63e91a0492/packages/core/utils/src/content-types.ts#L190-L212